### DISCUSSION TITLE
Tauri Signer Workspace Not Found

### DISCUSSION THREAD

**Discussion by elanzini** [tauri-apps/tauri#5754] [Q&A] [answered]
(1) I am trying to use the Tauri CLI to sign updates as described [here](https://tauri.app/v1/guides/distribution/updater/#signing-updates).
I am running: `npm run tauri signer generate -w ~/.tauri/myapp.key`
And I am getting this error:
```
npm ERR! No workspaces found:
npm ERR!   --workspace=/Users/edo/.tauri/myapp.key
```
(The folder `/Users/edo/.tauri`) exists.

The CLI works without the `-w` flag but I was wondering what a `workspace` is and if I should create one.

(2) Also, [here](https://tauri.app/v1/guides/distribution/updater/#update-server-json-format) it mentions that the `signature` parameter is necessary and that:
```
"signature" is the content of the .sig file which was generated by Tauri's CLI.
```
But no `.sig` file is generated by the Tauri CLI without the `-w` argument.

(3) Is it possible to use the Updater functionality without having Code signed the application?

**Comment by Simo555-web** [comment]
import urllib.request, json
resp = urllib.request.urlopen('
https://query2.finance.yahoo.com/v10/finance/quoteSummary/tsla?modules=price
')
data = json.loads(resp.read())
price =
data['quoteSummary']['result'][0]['price']['regularMarketPrice']['raw']
print(price)

**Comment by FabianLars** [comment, marked as answer]
1: You need a set of `--` to tell npm to forward the args to Tauri:
```sh
npm run tauri signer generate -- -w ~/.tauri/myapp.key
```
p.s. i'm rewriting the guide rn so that will be clearer soon.

2: Once you set the keys and the updater up it will generate the sig files.

3: Yes, the .sig files and tauri keys are not related to code signing. It's just to verify the updater bundles after downloading them. If you actually mean this and not real code signing then no, it's required.

**Reply by elanzini** [reply]
Thanks @FabianLars!

Where exactly is the `.sig` file expected to be generated? I see a `myapp.key` and `myapp.key.pub` files being generated, but no `.sig` file in the folder. Is the `.sig` file generated for each version? How to do that?

**Reply by FabianLars** [reply]
The .sig file is generated on `tauri build`, it changes on each change. The file will be next to the bundle, see https://tauri.app/v1/guides/distribution/updater#bundler-artifacts

**Reply by elanzini** [reply]
That's clear thanks. I followed the docs [here](https://tauri.app/v1/guides/distribution/updater/#javascript-api) to check for updates and on my server, I return `204` with No Content is the user is running on the latest version, as detailed [here](https://tauri.app/v1/guides/distribution/updater/#server-support).

Nonetheless, I am seeing `Tauri API error: Network Error: Json Error: EOF while parsing a value at line 1 column 0` being thrown by the `checkUpdate` function.

The code on my express server backend just does ` res.status(204).send();` in case the version of the app from the user matches with the latest.

**Reply by elanzini** [reply]
I got a bit further and I am now getting this when it tries to update:
```
thread 'tokio-runtime-worker' panicked at 'application failed to start: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /Users/edo/.cargo/registry/src/github.com-1ecc6299db9ec823/tauri-1.0.5/src/api/process.rs:87:8
thread 'tokio-runtime-worker' panicked at 'application failed to start: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /Users/edo/.cargo/registry/src/github.com-1ecc6299db9ec823/tauri-1.0.5/src/api/process.rs:87:8
```
Have you seen this before?

**Reply by FabianLars** [reply]
Not really no. Checking the source code it's where the [`restart` function](https://docs.rs/tauri/latest/tauri/api/process/fn.restart.html) is declared. https://github.com/tauri-apps/tauri/blob/1.0/core/tauri/src/api/process.rs#L87

no idea what the issue is, when does that error happen?

**Reply by elanzini** [reply]
I think it was happening because I was running in `dev`.
